### PR TITLE
fix french possessive determiner in case of vowel hiatus

### PR DESCRIPTION
**Pull request by blegaut**
Thanks for contributing!

## Please check if the PR fulfills these requirements 👌:

- [x] Confirm the PR related to a dedicated issue
- [x] Confirm your PR corrects a single bug or implements a single feature
- [x] Your code is linted locally prior to submission
- [x] Your code is fully tested: 99% to 100% coverage
- [ ] Everything is correct on [Sonar dashboard](https://sonarcloud.io/dashboard?id=RosaeNLG_RosaeNLG)
- [x] Confirm your code is under Apache 2.0 license
- [x] Confirm your documentation is under CC-BY-4.0 license
- [x] Confirm license and copyright content is created/updated in each file (see `CONTRIBUTING.md`)
- [x] Confirm `changelog.adoc` is updated
- [ ] If corrects vulnerabilities, confirm `changelog.adoc` contains CVE IDs

Also remember: each commit **MUST** contain a sign off message (see `CONTRIBUTING.md`) 🙄

## Indicate related issue: 

issue #180

### CODE REVIEW

File: packages/french-determiners/src/index.ts [hunk 1]
```diff
@@ -23,6 +23,20 @@ type getDetParameters = {
...
+  ma, ta, sa deviennent mon, ton, son devant des noms féminins commençant par une voyelle pour éviter le hiatus de deux voyelles. 
+  a priori, ce n'est pas le cas avec y: par exemple ma yogourtière
+  https://open.byu.edu/grammaire_ouverte/determinants#:~:text=N'oubliez%20pas%20de%20faire,le%20hiatus%20de%20deux%20voyelles.
+*/
+function getFixHiatus(contentAfterDet: string) {
+  if (!contentAfterDet) {
+    return false;
+  }
+  const firstLetterAfter = contentAfterDet.replace(/¤/g, ' ').trim().substring(0, 1);
+  const fixHiatus = ['a', 'e', 'é', 'è', 'ê', 'i', 'o', 'u'].includes(firstLetterAfter);
```
Comment (ludans) [inline packages/french-determiners/src/index.ts:36]: Can you use the full list? àáâãäåÀÁÂèéêëÈÉÊËìíîïÌÍÎÏòóôõöøÒÓÔÕÖØùúûüÙÚÛÜ

(we will barely find a word starting with ø in French, but capital letters can happen)

File: packages/french-determiners/src/index.ts [hunk 2]
```diff
@@ -23,6 +23,20 @@ type getDetParameters = {
   forceDes?: boolean;
 };
 
+/* 
+  ma, ta, sa deviennent mon, ton, son devant des noms féminins commençant par une voyelle pour éviter le hiatus de deux voyelles. 
+  a priori, ce n'est pas le cas avec y: par exemple ma yogourtière
+  https://open.byu.edu/grammaire_ouverte/determinants#:~:text=N'oubliez%20pas%20de%20faire,le%20hiatus%20de%20deux%20voyelles.
+*/
+function getFixHiatus(contentAfterDet: string) {
```
Comment (ludans) [inline packages/french-determiners/src/index.ts:31]: suggest to rename in `mustFixHiatus` if responding with a boolean?
a function called `getFixHiatus` should take 3 parameters: contentAfterDet, classicOption (e.g. ma), hiatusOption (e.g. mon)

**Review by ludans — CHANGES_REQUESTED** [review]
Thanks a lot for the PR.
I didn't even realize the existence of hiatus.

Can you please check the comments?

File: packages/french-determiners/src/index.ts (hunk 1 above)
Comment (blegaut) [reply packages/french-determiners/src/index.ts]: I also add support for the h muet

File: packages/french-determiners/src/index.ts (hunk 2 above)
Comment (blegaut) [reply packages/french-determiners/src/index.ts]: done

File: packages/french-determiners/src/index.ts [hunk 3]
```diff
@@ -23,6 +23,20 @@ type getDetParameters = {
...
+/* 
+  ma, ta, sa deviennent mon, ton, son devant des noms féminins commençant par une voyelle pour éviter le hiatus de deux voyelles. 
+  a priori, ce n'est pas le cas avec y: par exemple ma yogourtière
+  https://open.byu.edu/grammaire_ouverte/determinants#:~:text=N'oubliez%20pas%20de%20faire,le%20hiatus%20de%20deux%20voyelles.
+*/
+function getFixHiatus(contentAfterDet: string) {
+  if (!contentAfterDet) {
+    return false;
+  }
+  const firstLetterAfter = contentAfterDet.replace(/¤/g, ' ').trim().substring(0, 1);
```
[the remark this answers is not in the document]
Comment (blegaut) [reply packages/french-determiners/src/index.ts]: I moved this part to the rosae-lib package

**Review by ludans — CHANGES_REQUESTED** [review]
Thanks for adding the h support.
There is one very last thing to change and I will merge.

**Review by ludans — APPROVED** [review]

**Pull request merged**